dangerfile.js

```diff
@@ -1,76 +1,97 @@
-        );
-    }
-}
+const PR_LINK = `([#${danger.github.pr.number}](${danger.github.pr.html_url}))`;
 
-async function checkChangelog() {
-    const changelogFile = "CHANGELOG.md";
+const CHANGELOG_SUMMARY_TITLE = `Instructions and example for changelog`;
+const CHANGELOG_BODY = `Please add an entry to \`CHANGELOG.md\` to the "Unreleased" section. Make sure the entry includes this PR's number.
 
-    // Check if skipped
-    const skipChangelog = danger.github && (danger.github.pr.body + "").includes("#skip-changelog");
-    if (skipChangelog) {
-        return;
-    }
+Example:`;
 
-    // Check if current PR has an entry in changelog
-    const changelogContents = await danger.github.utils.fileContents(changelogFile);
+const CHANGELOG_END_BODY = `If none of the above apply, you can opt out of this check by adding \`#skip-changelog\` to the PR description.`;
 
-    const hasChangelogEntry = RegExp(`#${danger.github.pr.number}\\b`).test(changelogContents);
-    if (hasChangelogEntry) {
-        return;
-    }
+function getCleanTitleWithPrLink() {
+  const title = danger.github.pr.title;
+  return title.split(": ").slice(-1)[0].trim().replace(/\.+$/, "") + PR_LINK;
+}
 
-    // Short-circuit if lacking permissions
-    const hasCommentPermission = danger.github.pr.head.repo.git_url == danger.github.pr.base.repo.git_url;
-    if (!hasCommentPermission) {
-        console.log("Please consider adding a changelog entry for the next release.");
-        process.exitCode = 1;
-        return;
-    }
+function getChangelogDetailsHtml() {
+  return `
+### ${CHANGELOG_SUMMARY_TITLE}
 
-    // Report missing changelog entry
-    fail(
-        "Please consider adding a changelog entry for the next release.",
-        changelogFile
-    );
+${CHANGELOG_BODY}
 
-    const prTitleFormatted = danger.github.pr.title
-        .split(": ")
-        .slice(-1)[0]
-        .trim()
-        .replace(/\.+$/, "");
+\`\`\`markdown
+- ${getCleanTitleWithPrLink()}
+\`\`\`
 
-    markdown(
-        `
-### Instructions and example for changelog
+${CHANGELOG_END_BODY}
+`;
+}
 
-Please add an entry to \`CHANGELOG.md\` to the "Unreleased" section. Make sure the entry includes this PR's number.
+function getChangelogDetailsTxt() {
+  return (
+    CHANGELOG_SUMMARY_TITLE +
+    "\n" +
+    CHANGELOG_BODY +
+    "\n" +
+    getCleanTitleWithPrLink() +
+    "\n" +
+    CHANGELOG_END_BODY
+  );
+}
 
-Example:
+function HasPermissionToComment() {
+  return (
+    danger.github.pr.head.repo.git_url == danger.github.pr.base.repo.git_url
+  );
+}
 
-\`\`\`markdown
-## Unreleased
+async function containsChangelog(path) {
+  const contents = await danger.github.utils.fileContents(path);
+  return contents.includes(PR_LINK);
+}
 
-- ${prTitleFormatted} ([#${danger.github.pr.number}](${danger.github.pr.html_url}))
-\`\`\`
+async function checkChangelog() {
+  const skipChangelog =
+    danger.github && (danger.github.pr.body + "").includes("#skip-changelog");
```

Comment:
It would be nice in the future that skip-change log would be more flexible, not being case sensitive and allowing the `_` symbol, so folks can write `#SKIP-CHANGELOG` or `#skip_changelog` but that could be done in another PR